client/src/components/utils/sasCode.ts

```diff
@@ -48,14 +49,23 @@ export function wrapCodeWithOutputHtml(code: string): string {
     const htmlStyleOption = generateHtmlStyleOption();
     return `title;footnote;ods _all_ close;
 ods graphics on;
-ods html5${htmlStyleOption} options(bitmap_mode='inline' svg_mode='inline');
+ods html5${htmlStyleOption} options(bitmap_mode='inline' svg_mode='inline') body="${v4()}.htm";
 ${code}
 ;*';*";*/;run;quit;ods html5 close;`;
   } else {
     return code;
   }
 }
 
+export function extractOutputHtmlFileName(
+  line: string,
+  defaultValue: string,
+): string {
+  return (
+    line.match(/body="(.{8}-.{4}-.{4}-.{4}-.{12}).htm"/)?.[1] ?? defaultValue
```

Comment:
This is probably ok for now, but at some point in the future, we should probably use the ODS IOM events instead (namely ODSComplete), similar to the way EG and SAS Studio handle. Example C# code:

ODS_1_1 _odsService = (ODS_1_1)ws.ODS;
CIODSFileEvents_1_1_Event _odsFileEventsFor93 = ws.ODS as CIODSFileEvents_1_1_Event;
_odsService.FileOpen += HandleOdsFileOpen;
_odsService.FileClose += HandleOdsFileClose;
_odsService.DirectoryBegin += HandleOdsDirectoryBegin;
_odsService.AnchorElement += HandleOdsAnchorElement;
_odsFileEventsFor93.ODSComplete += HandleOdsComplete;

Reply:
Hey @clangsmith . Thanks for this. I linked this comment to the IOM library panel issue and will take a look when I'm working on that.